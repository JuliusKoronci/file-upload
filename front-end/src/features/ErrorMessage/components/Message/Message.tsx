import { SFC } from 'react';
import * as React from 'react';

import { Chip } from '@material-ui/core';
import { Error } from '@material-ui/icons';

interface IMessageProps {
  message?: string
}

/**
 * Display an error message if set
 */
export const Message: SFC<IMessageProps> = ({ message }) => {
  if (!message) {
    return null;
  }
  return (<Chip
    icon={<Error />}
    label={message}
    color="secondary"
  />)
};