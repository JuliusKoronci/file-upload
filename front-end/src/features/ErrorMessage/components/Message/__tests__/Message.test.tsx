import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Message } from '../Message';


describe('Message', () => {
  it('should render snapshot', () => {
    expect(toJson(shallow(<Message
      message="my awesome message"
    />))).toMatchSnapshot();
  });
  it('should render snapshot without a message', () => {
    expect(toJson(shallow(<Message />))).toMatchSnapshot();
  });
});