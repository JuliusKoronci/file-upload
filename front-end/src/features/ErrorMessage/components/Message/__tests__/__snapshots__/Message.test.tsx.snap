// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should render snapshot 1`] = `
<WithStyles(Chip)
  color="secondary"
  icon={<pure(ErrorIcon) />}
  label="my awesome message"
/>
`;

exports[`Message should render snapshot without a message 1`] = `""`;
